trillian/migrillian/core/controller.go

```diff
@@ -207,6 +207,19 @@ func (c *Controller) RunWhenMaster(ctx context.Context) error {
 	}
 }
 
+// runWithRestarts calls Run until it succeeds or the context is done.
+func (c *Controller) runWithRestarts(ctx context.Context) error {
+	for ctx.Err() == nil {
+		if err := c.Run(ctx); err != nil { // There was an error, so we try again.
+			glog.Errorf("%s: Controller.Run: %v", c.label, err)
```

Comment:
Is it correct for non continuous runs to retry here?

Reply:
Don't know what is best. If, say, the non-continuous run must copy 1M certs, but the source log flakes after 500k, then should we fail or retry?

Prob. it's best to make the non-continuous mode as simple as possible, as it's only used for one-off migrations. So, not restarting is better, I will update the code.

Reply:
Done.